File: backend/goat_listing/src/main/java/com/goat/cart/domain/exceptions/ListingNotAvailableException.java
package com.goat.cart.domain.exceptions;

/**
 * Excepción lanzada cuando el listing no está disponible (no es PUBLISHED).
 */
public class ListingNotAvailableException extends RuntimeException {
    public ListingNotAvailableException(String message) {
        super(message);
    }

    public ListingNotAvailableException(String message, Throwable cause) {
        super(message, cause);
    }
}

File: backend/goat_catalog/src/goat_catalog/cart/domain/exceptions/listing_not_available_exception.py
"""Excepción cuando un listing no está disponible."""


class ListingNotAvailableException(Exception):
    """Excepción lanzada cuando un listing no está disponible para agregar al carrito."""

    def __init__(self, message: str = "El listing no está disponible") -> None:
        """Inicializa la excepción.

        Args:
            message: Mensaje de error
        """
        super().__init__(message)
        self.message = message

